schemas/tls.yml

```diff
@@ -0,0 +1,174 @@
+---
+- name: tls
+  title: TLS
+  group: 2
+  short: Fields describing a TLS connection.
+  description: >
+    Fields related to TLS activity.
+  type: group
+  fields:
+
+    - name: version.protocol
+      level: extended
+      type: keyword
+      description: Normalized protocol name parsed from original string (e.g. ssl, tls).
+
+    - name: version.number
+      level: extended
+      type: keyword
+      description: Numeric part of the version parsed from the original string (e.g. 1.2, 3).
```

Comment:
These two fields cannot be nested under `tls.version`, as this is a hard break vs Packetbeat's current definition for `tls.version`, where the field is type `keyword`. This would lead to a mapping conflict within the 7.x product line, which we can't do.

What would work well, I think is the following:

- `tls.version` (`keyword`)
  - Field content would need to move from Packetbeat's content of e.g. "TLS 1.3" to "1.3", however
  - perfectly aligned with all other ECS `*.version` fields.
- `tls.version_protocol` (`keyword`)
  - This is where we would put either "tls" or "ssl"
  - The underscore avoids the mapping conflict
  - The definition of `tls.version_protocol` should mention that the protocol must be lowercased, in order to be aligned with network fields such as `network.protocol`, `network.application`, `network.type` and `network.transport`.

What I'm laying out above however still leads to a minor break within the 7.x product line, where Packetbeat would have to start putting "pure" version values such as "1.2", "1.3" instead of its current "TLS 1.2", "TLS 1.3".

I'd like to get people's take on that kind of break for the `tls.version` field's content for Packetbeat. @andrewkroh @tsg @ruflin WDYT?

Reply:
My review had been in progress while you moved things around for this part of the file, but this comment is still applicable after your recent push.

We can't nest under `tls.version.*` as this is a hard breaking change vs Packetbeat.

Reply:
I don't think that really _breaks_ the version string. The only thing that is broken is doing analysis across agent versions. Even then, analysis will generally still work, but be degraded, which seems like a reasonable progression.

Reply:
That's also my take. I hope we can take this step forward. But I also feel it's important we get alignment from folks on this :-)